nilearn/input_data/nifti_spheres_masker.py

```diff
@@ -128,9 +136,18 @@ def _get_spheres_rows(seeds, target_affine, target_shape, radius, allow_overlap)
 
     # Include the voxel containing the seed itself
     brain_coords = brain_coords.astype(int).tolist()
-    for i, seed in enumerate(seeds):
-        adjacency[i, brain_coords.index(seed)] = True
+    # Get the mask indices (in terms of brain coords)
+    mask_idx = [brain_coords.index(i) for i in mask_coords.tolist()]
 
+    for i, seed in enumerate(seeds):
+        try:
+            adjacency[i, brain_coords.index(seed)] = True
```

Comment:
As seeds are not integer, but the brain_coords are, the try - catch is needed here. 

Maybe converting `seed` here to `int` would solve this?